kernel/src/hil/flash.rs

```diff
@@ -125,3 +127,36 @@ pub trait Client<F: Flash> {
     /// Flash erase complete.
     fn erase_complete(&self, error: Error);
 }
+
+pub trait FlashInfo {
+    fn flash_size(&self) -> u32;
+    fn num_pages(&self) -> u32;
+    fn page_size(&self) -> u32;
+    /// Return 0 if flash does not support locking
+    fn num_lock_units(&self) -> u32;
+    /// Return 0 if flash does not support locking
+    fn lock_unit_size(&self) -> u32;
+    /// Return 0 if flash does not support locking
+    fn pages_per_lock_unit(&self) -> u32;
+    fn page_to_lock_unit(&self, page: u32) -> u32;
+}
+
+pub trait FlashLocking {
```

Comment:
Also, do we expect Flash Locking to be a capability that all Flash hardware supports?